video/gazebocamerainput.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+// BoB robotics includes
+#include "common/assert.h"
+#include "input.h"
+
+// Gazebo includes
+#include <gazebo/gazebo_config.h>
+#include <gazebo/transport/transport.hh>
+#include <gazebo/msgs/msgs.hh>
+#include <gazebo/common/common.hh>
+
+// Gazebo's API has changed between major releases. These changes are
+// accounted for with #if..#endif blocks in this file.
+#if GAZEBO_MAJOR_VERSION < 6
+#include <gazebo/gazebo.hh>
+#else
+#include <gazebo/gazebo_client.hh>
+#endif
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+// Standard C++ includes
+#include <stdexcept>
+#include <mutex>
+
+namespace BoBRobotics {
+namespace Video {
+
+constexpr const char *GazeboCameraDeviceName = "Gazebo Camera";
+
+//----------------------------------------------------------------------------
+// BoBRobotics::Video::GazeboCameraInput
+//----------------------------------------------------------------------------
+//! A thin wrapper for reading from any video source supported by OpenCV
+class GazeboCameraInput : public Input
+{
+public:
+    //! Create a new video stream
+    GazeboCameraInput()
+      : GazeboCameraInput()
+    {}
```

Comment:
I've changed this API quite a bit in the latest commit. This zero-arg constructor is not there anymore. However, I've added a constructor which creates a Gazebo transport node in case the user does not supply one.

Reply:
Cool. That's a sensible idea.